File: 백준/Silver/1475. 방 번호/방 번호.py
import sys
fi = sys.stdin.readline
                
def solution():
    n = fi().rstrip()
    res = [0] * 10
    
    for i in n:
        if int(i) == 6 or int(i) == 9:
            if res[6] > res[9]: res[9] += 1
            else: res[6] += 1
        else:
            res[int(i)] += 1
        
    print(max(res))
        
    
if __name__ == '__main__':
    solution()

File: 백준/Silver/1475. 방 번호/방 번호.java
import java.io.*;
import java.util.*;

public class Main {
    public static void solution() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        
        int[] arr = new int[10];
        int n = Integer.parseInt(br.readLine());
        int ans = 0;
        
        while(n > 0) {
            arr[n%10]++;
            n /= 10;
        }
        for(int i=0;i<10;i++) {
            if(i==6 || i==9) continue;
            ans = Math.max(ans, arr[i]);
        }
        ans = Math.max(ans, (arr[6]+arr[9]+1)/2);
        sb.append(ans);
        
        System.out.print(sb);
        br.close();
    }
    
    public static void main(String[] args) throws IOException {
        Main.solution();
    }
}